File: src/main/java/nl/haltedata/netex/dto/NetexQuayInSequence.java
package nl.haltedata.netex.dto;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "v_netex_quay_in_sequence")
@Getter
@Setter
@IdClass(NetexQuayInSequenceId.class)
public class NetexQuayInSequence {
    @Id
    private Long sequenceId;
    @Id
    private Long quayIndex;
    private String lineNumber;
    private String quayCode;
    private String quayName;
    private String stopSideCode;
    private String stopPlaceCode;
    private String quayLocationType;
}

File: frontend/src/app/routes/netex-quay-in-sequence.ts
export class NetexQuayInSequence {
    public sequenceId : Number;
    public quayIndex : Number;
    public lineNumber : String;
    public quayCode : String;
    public quayName : String;
    public stopSideCode : String;
    public stopPlaceCode : String;
    public quayLocationType : String;

    constructor(quay: NetexQuayInSequence) {
        this.sequenceId = quay.sequenceId;
        this.quayIndex = quay.quayIndex;
        this.lineNumber = quay.lineNumber;
        this.quayCode = quay.quayCode;
        this.quayName = quay.quayName;
        this.stopSideCode = quay.stopSideCode;
        this.stopPlaceCode = quay.stopPlaceCode;
        this.quayLocationType = quay.quayLocationType;
    }
    
    public get name(): string {
        return this.quayName + (this.stopSideCode ? " (" + this.stopSideCode + ")" : "");
    }
}
